framework/db/plugin_uploader.py

```diff
@@ -0,0 +1,143 @@
+from __future__ import print_function
+from sys import getsizeof, stderr
+from itertools import chain
+from collections import deque
+import os
+from datetime import datetime
+import re
+
+from ptp import PTP
+from framework.dependency_management.dependency_resolver import ServiceLocator
+from framework.db import models
+from framework.db.target_manager import target_required
+
+class PluginUploader():
+
+	def __init__(self, tool_name):
+		self.supported_tools = ['w3af', 'skipfish', 'arachni', 'hoppy', 'burpsuite']
+		#self.full_parse_tools = ['w3af', 'skipfish', 'arachni']
+		self.db = ServiceLocator.get_component("db")
+		self.transaction = ServiceLocator.get_component("transaction")
+		self.re_extras = re.compile(r"(.*?) (\/.*?||\w+.*?) HT.*?")
+		if tool_name in self.supported_tools:
+			self.tool_name = tool_name
+			self.ptp = PTP(tool_name)
+		else:
+			self.tool_name = None
+			raise Exception("%s tool not supported by this plugin" % tool_name)
+
+	def init_uploader(self, pathname):
+		parsed_data = self.ptp.parse(pathname)
+		self.parsed_data = parsed_data[-1]['transactions']
+
+	@target_required
+	def OWTFDBUpload(self, target_id=None):
+		if(target_id is None):
+			print("Target id is None, aborting uploader!!")
+			return
+		if self.parsed_data is None:
+			print("Nothing found in the file, aborting uploader!!")
+			return
+		self.upload_checks()
+		target_url = ServiceLocator.get_component("target").GetTargetURLForID(target_id)
+		for data in self.parsed_data:
+			extras = self.re_extras.findall(data['request'])[0]
+			Method = extras[0]
+			Url = target_url + extras[1]
+			# Converting to latin1 to avoid ascii encoding errors
+			if self.tool_name == 'burpsuite': data['body'] = data['body'].encode('latin1', 'ignore')
+			transaction_model = models.Transaction(
+				url=Url,
+				scope=None,
+				method=Method,
+				data=None,
+				time=None,
+				time_human="Uploaded by " + self.tool_name, # to distinguish uploader reports in transaction table
```

Comment:
I have no preference. You tell me how to make it different when transactions are uploaded instead of proxying?